lib/screens/qr_code_dialog.dart

```diff
@@ -35,19 +46,68 @@ class QrCodeDialog extends StatelessWidget {
                 ],
               );
             }
-            final content =
-                snapshot.data!.reduce((value, element) => '$value,$element');
-
-            return Center(
-              child: SizedBox(
-                width: smallestSide,
-                height: smallestSide,
-                child: QrImage(
-                  data: content,
-                  version: QrVersions.auto,
+            if (snapshot.connectionState == ConnectionState.waiting ||
+                !snapshot.hasData) {
+              return Center(child: CircularProgressIndicator());
+            }
+
+            final storedDeviceName = snapshot.data![0] as String;
+            final storedIp = snapshot.data![1] as String?;
+            final storedPort = snapshot.data![2] as int;
+
+            final ips = networkInfoService.ips;
+            final selectedIp = _selectIp(ips, storedIp);
+            if (selectedIp != storedIp && selectedIp != null) {
+              identityService.setIp(selectedIp);
+            }
+
+            return Column(children: [
+              if (!peerService.isServerRunning)
```

Comment:
If the QR Code cannot be opened without a running server (see comment above), this check may be removed.